### PR TITLE
Data layer new protocol added

### PR DESCRIPTION
**Pull request by jeffh1996**
Added the new protocol in the data layer but kept the old protocol (Will need to make a new PR(s) to change to the new protocol since many parts of the program will be affected and I do not want to make this PR too large).

Note: Since CMUs are non-existent anymore, this might also need to be part of a new PR since a lot of files are affected.

### CODE REVIEW

File: src/DataLayer/BatteryData.h [hunk 1]
```diff
@@ -95,4 +95,85 @@ struct BatteryData
...
+    unsigned char fanSpeed;
+    unsigned char requestedFanSpeed;
+    unsigned short int lowCellVoltage;
+    unsigned char lowCellVoltageId;
+    unsigned short int highCellVoltage;
+    unsigned char highCellVoltageId;
+    unsigned short int averageCellVoltage;
+    PrechargeState prechargeState;
+    const char* prechargeStateJSON;
+    unsigned char auxVoltage;
```
Comment (enochtsang) [inline src/DataLayer/BatteryData.h]: Why is it all just added commented code, why isn't it actually changed?

File: src/DataLayer/CcsDefines.h [hunk 2]
```diff
@@ -10,12 +10,12 @@ namespace CcsDefines
...
     const unsigned char BATTERY_FAULTS_PKG_ID = 6;
     const unsigned char BATTERY_PKG_ID = 7;
-    const unsigned char CMU_PKG_ID = 8;
+    const unsigned char CMU_PKG_ID = 8;		//TODO:Delete all references to CMUs
     const unsigned char MPPT_PKG_ID = 9;
     const unsigned char LIGHTS_PKG_ID = 10;
 
     const int MPPT_COUNT = 3;
-    const int CMU_COUNT = 4;
+    const int CMU_COUNT = 4;				//TODO:Delete all references to CMUs
```
Comment (enochtsang) [inline src/DataLayer/CcsDefines.h]: Not TODO, it's part of this PR.

**Comment by jeffh1996** [comment]
Will reopen after making some changes

File: src/DataLayer/CcsDefines.h [hunk 3]
```diff
@@ -30,4 +29,7 @@ namespace CcsDefines
     // These number masks out all the errors that don't really matter
     const unsigned int MAJOR_BMU_ERROR_MASK = 0x1C37;
     const unsigned int MAJOR_MC_ERROR_MASK = 0xFF;
+
+    //No data
+    const bool NO_DATA = 0;
```
Comment (enochtsang) [inline src/DataLayer/CcsDefines.h]: I like it, but maybe leave a comment on why this exists. The "No data" comment isn't helpful at all.

**Comment by jeffh1996** [comment]
Added a description for the NO_DATA member and added some data checks in the tests to ensure that the data members before and after the NO_DATA members are correct.

As for the CMU files. They should have already been removed in the previous PR and when I updated this branch in 2eafa6abf5aef63209b6d1e56d78500194c73b4e .

**Pull request merged**